_episodes/03-intro-git-github.md

```diff
@@ -1214,1004 +1218,1018 @@ repository (`git commit`):
-Collaborator.
+> ## Don't Lose Your HEAD
+>
+> Above we used
+>
+> ```
+> $ git checkout f22b25e notes.txt
+> ```
+> {: .language-bash}
+>
+> to revert `notes.txt` to its state after the commit `f22b25e`. But be careful!
+> The command `checkout` has other important functionalities and Git will misunderstand
+> your intentions if you are not accurate with the typing. For example,
+> if you forget `notes.txt` in the previous command.
+>
+> ```
+> $ git checkout f22b25e
+> ```
+> {: .language-bash}
+> ```
+> Note: checking out 'f22b25e'.
+>
+> You are in 'detached HEAD' state. You can look around, make experimental
+> changes and commit them, and you can discard any commits you make in this
+> state without impacting any branches by performing another checkout.
+>
+> If you want to create a new branch to retain commits you create, you may
+> do so (now or later) by using -b with the checkout command again. Example:
+>
+>  git checkout -b <new-branch-name>
+>
+> HEAD is now at f22b25e Make a change that I'll regret later
+> ```
+> {: .error}
+>
+> The "detached HEAD" is like "look, but don't touch" here,
+> so you shouldn't make any changes in this state.
+> After investigating your repo's past state, reattach your `HEAD` with `git checkout master`.
+{: .callout}
 
-To download the Collaborator's changes from GitHub, the Owner now enters:
+It's important to remember that
+we must use the commit number that identifies the state of the repository
+*before* the change we're trying to undo.
+A common mistake is to use the number of
+the commit in which we made the change we're trying to discard.
+In the example below, we want to retrieve the state from before the most
+recent commit (`HEAD~1`), which is commit `f22b25e`:
 
-```
-$ git pull origin master
-```
-{: .language-bash}
+![Git Checkout]({{ page.root }}/fig/git/git-checkout.svg)
 
-```
-remote: Enumerating objects: 4, done.
-remote: Counting objects: 100% (4/4), done.
-remote: Compressing objects: 100% (2/2), done.
-remote: Total 3 (delta 0), reused 3 (delta 0), pack-reused 0
-Unpacking objects: 100% (3/3), done.
-From https://github.com/USERNAME/un-report
- * branch            master     -> FETCH_HEAD
-   9272da5..29aba7c  master     -> origin/master
-Updating 9272da5..29aba7c
-Fast-forward
- notes.txt | 1 +
- 1 file changed, 1 insertion(+)
- create mode 100644 notes.txt
-```
-{: .output}
+We have now reverted our current file and commit to the latest version without the bug. But we have kept the commit and history from the commit that had the error.
 
-Now the three repositories (Owner's local, Collaborator's local, and Owner's on
-GitHub) are back in sync!
+In the next section, we will learn how to share our git repository across networks. We can do this by creating a link to another location, like Github (remote repository) which we can `push` and `pull` our repository to and from to share our files with others as well as backing up our local repository.
```

Comment:
Old code for a section that's already been covered. Move or delete?

Reply:
decided to delete it (commit 9facef1)